File: artifacts/tulipindicators/decompose/src/todeg.c
int ti_todeg_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_todeg(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *in1 = inputs[0];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = in1[i] * (180.0 / 3.14159265358979323846);
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/translate/src/todeg.rs
use crate::*;

pub fn ti_todeg_start(options: Option<&f64>) -> i32 {
    let _ = options; // Explicitly ignore the parameter to match C behavior
    0
}
pub fn ti_todeg(
    size: i32,
    inputs: &[&[f64]],
    options: &[f64],
    outputs: &mut [&mut [f64]],
) -> i32 {
    let in1 = inputs[0];
    let output = &mut outputs[0];
    
    for i in 0..size as usize {
        output[i] = in1[i] * (180.0 / std::f64::consts::PI);
    }
    
    0
}
